core/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequest.java

```diff
@@ -481,16 +481,28 @@ public RestoreSnapshotRequest source(Map<String, Object> source) {
                     throw new IllegalArgumentException("malformed indices section, should be an array of strings");
                 }
             } else if (name.equals("partial")) {
-                partial(lenientNodeBooleanValue(entry.getValue()));
+                try {
+                    partial(nodeBooleanValue(entry.getValue()));
+                } catch (IllegalArgumentException ex) {
+                    throw new IllegalArgumentException("Could not convert [partial] to boolean.", ex);
```

Comment:
I wonder if it is worth having a version of `nodeBooleanValue` that tasks a string for the name of the field being converted. I know we have a few methods like that in other places.